### ISSUE TITLE
compilation failing in 2.12.x community build 

### ISSUE THREAD

**Issue by SethTisue** [non/spire#592]
at https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/841/consoleFull and other recent runs we see:

```
[spire] [warn] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/spire-f6e9c8eaf2a524ddab87f0131380246ca2810d04/core/shared/src/main/scala/spire/math/Algebraic.scala:1300: discarding unmoored doc comment
[spire] [warn]       /** Upper bound on the value. */
[spire] [warn]       ^
[spire] [info] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/spire-f6e9c8eaf2a524ddab87f0131380246ca2810d04/core/shared/src/main/scala/spire/optional/mapIntIntPermutation.scala:39: non-literal range
[spire] [info]     cforRange(0 until sa.size) { k =>
[spire] [info]                                ^
[spire] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/spire-f6e9c8eaf2a524ddab87f0131380246ca2810d04/core/shared/src/main/scala/spire/std/option.scala:12: value combine is not a member of spire.algebra.Semigroup[A]
[spire] [error]     case (Some(x), Some(y)) => Some(x |+| y)
[spire] [error]                                       ^
[spire] [info] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/spire-f6e9c8eaf2a524ddab87f0131380246ca2810d04/core/shared/src/main/scala/spire/optional/mapIntIntPermutation.scala:39: non-literal range
[spire] [info]     cforRange(0 until sa.size) { k =>
[spire] [info]                                ^
[spire] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/spire-f6e9c8eaf2a524ddab87f0131380246ca2810d04/core/shared/src/main/scala/spire/syntax/std/Ops.scala:91: too many arguments (2) for method combine: (as: TraversableOnce[A])A
[spire] [error]     cfor(0)(_ < arr.length, _ + 1) { i => result |+|= arr(i) }
[spire] [error]                                                          ^
[spire] [warn] one warning found
[spire] [error] two errors found
[spire] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/spire-f6e9c8eaf2a524ddab87f0131380246ca2810d04/core/shared/src/main/scala/spire/std/option.scala:12: value combine is not a member of spire.algebra.Semigroup[A]
[spire] [error]     case (Some(x), Some(y)) => Some(x |+| y)
[spire] [error]                                       ^
[spire] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/spire-f6e9c8eaf2a524ddab87f0131380246ca2810d04/core/shared/src/main/scala/spire/syntax/std/Ops.scala:91: too many arguments (2) for method combine: (as: TraversableOnce[A])A
[spire] [error]     cfor(0)(_ < arr.length, _ + 1) { i => result |+|= arr(i) }
[spire] [error]                                                          ^
[spire] [info] No documentation generated with unsuccessful compiler run
[spire] [warn] two warnings found
[spire] [error] two errors found
[spire] [error] (coreJVM/compile:doc) Scaladoc generation failed
[spire] [error] (coreJVM/compile:compileIncremental) Compilation failed
```

I will deal with it at our end by disabling Scaladoc generation

**Comment by SethTisue** [comment]
hmm, it's not just a Scaladoc thing:

```
[spire] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/spire-d8f6e614fd74adc48c895ac8b66423adda67bcdf/core/shared/src/main/scala/spire/std/option.scala:12: value combine is not a member of spire.algebra.Semigroup[A]
[spire] [error]     case (Some(x), Some(y)) => Some(x |+| y)
[spire] [error]                                       ^
[spire] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/spire-d8f6e614fd74adc48c895ac8b66423adda67bcdf/core/shared/src/main/scala/spire/syntax/std/Ops.scala:91: too many arguments (2) for method combine: (as: TraversableOnce[A])A
[spire] [error]     cfor(0)(_ < arr.length, _ + 1) { i => result |+|= arr(i) }
[spire] [error]                                                          ^
[spire] [error] two errors found
```

from https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/843/consoleFull

**Comment by SethTisue** [comment]
I wonder if https://github.com/non/spire/pull/588 is actually broken on 2.12 but since y'all aren't really doing 2.12 yet, you wouldn't have noticed in your own CI. maybe rewinding to 8cd85fe328797b28b2cb64b5c9a265fe64d0761a will fix it (I have a hunch it will since that's where the 2.12.0 community build is frozen and that hasn't failed)

**Comment by SethTisue** [comment]
hmm, no, wrong guess, same error. I'll explore the possibility this is a real regression caused by something that was merged into Scala 2.12.x

**Comment by SethTisue** [comment]
I'm seeing the failure even with Scala `2.12.0-ad2f8ee-nightly` and `14c525fbe90958b1cc110bbfaaf199ea78a82c1c`, so that leaves me baffled given that the same versions passed at https://scala-ci.typesafe.com/view/scala-2.12.0/job/scala-2.12.0-integrate-community-build/23/consoleFull

I see two basic possibilities here:

1) the actual root of the problem is one of Spire's dependencies — this would account for why there's success in the 2.12.0 community build, where all projects are frozen, but failure in the 2.12.x community build, where various dependencies have moved on
2) my experiments were insufficiently careful

I don't have time to investigate further on my own. perhaps you'll hit this yourself once you attempt to publish for 2.12.0-RC2 and/or 2.12.0 final, and we'll find out more then.

provisionally closing.

**Comment by xuwei-k** [comment]
> [spire] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/spire-f6e9c8eaf2a524ddab87f0131380246ca2810d04/core/shared/src/main/scala/spire/std/option.scala:12: value combine is not a member of spire.algebra.Semigroup[A]
[spire] [error]     case (Some(x), Some(y)) => Some(x |+| y)
[spire] [error]                                       ^


https://github.com/typelevel/machinist/commit/86dd0e4ff619b7871c7c54670

- use older version machinist in community build
- or update latest machinist in spire https://github.com/non/spire/pull/600

**Comment by denisrosset** [comment]
This is because we are in the middle of a basing of spire on algebra, and the `Group` methods have been renamed in `cats-kernel`, in `machinist` but not on `spire`'s master.

This is corrected by #589 which I am waiting approval for.

**Comment by denisrosset** [comment]
Ok, for an interim release, see #606.
